var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -72,6 +73,13 @@ class Mpich(AutotoolsPackage):
         'mpicc', 'mpicxx', 'mpif77', 'mpif90', 'mpifort', relative_root='bin'
     )
 
+    # Fix using an external hwloc
+    # See https://github.com/pmodels/mpich/issues/4038
+    # and https://github.com/pmodels/mpich/pull/3540
+    patch('https://github.com/pmodels/mpich/commit/86ee1097d4507154cd18f8d8f5fcc0c53f106e11.patch',
+          sha256='27a339d9a879440cc37e1a89680a289d2c06797953fa226045b32d9247b8a1ea',
+          when='+hwloc')
```

Comment:
```suggestion
    patch('https://github.com/pmodels/mpich/commit/8a851b317ee57366cd15f4f28842063d8eff4483.patch',
          sha256='eb982de3366d48cbc55eb5e0df43373a45d9f51df208abf0835a72dc6c0b4774',
          when='@3.3 +hwloc')
```
I would use the commit that is on master just to be on the safe side. I'm not sure whether GitHub will clean up the PR commit at some point. Also, the patch does not apply to develop (the commit will be included in 3.4) or anything before 3.3.